src/main/java/com/springweb/dv_spring_web_mongo/service/ProjectService.java

```diff
@@ -26,25 +24,37 @@
         for (Project project : list) {
             listWithDto.add(project.convertToDTO());
         }
-        return listWithDto;
+        if (listWithDto.isEmpty()) throw new ProjectNotFoundException("Database is empty");
+        else return listWithDto;
     }
 
     public ProjectDTO getProjectById(String id) {
-        Optional<Project> project = projectRepository.findById(id);
-        return project.get().convertToDTO();
+        Optional<Project> optional = projectRepository.findById(id);
+        if (optional.isEmpty()) throw new ProjectNotFoundException("Project with id '" + id + "' not found");
+        else return optional.get().convertToDTO();
     }
 
     public void addNewProject(ProjectDTO projectDTO) {
-        projectRepository.save(projectDTO.convertToProject());
+        if (projectDTO.getProjectName() == null || projectDTO.getProjectName().isEmpty())
```

Comment:
Please try to resolve the issue using following approach:
https://www.javaguides.net/2021/04/spring-boot-dto-validation-example.html